src/resolv/async_resolv.c

```diff
@@ -1468,38 +1535,71 @@ resolv_get_string_ptr_address(TALLOC_CTX *mem_ctx,
     return straddr;
 }
 
-struct sockaddr_storage *
+struct sockaddr *
 resolv_get_sockaddr_address_index(TALLOC_CTX *mem_ctx,
                                   struct resolv_hostent *hostent,
-                                  int port, int addrindex)
+                                  int port, int addrindex,
+                                  socklen_t *sockaddr_len)
 {
-    struct sockaddr_storage *sockaddr;
+    struct sockaddr *sockaddr;
+    int len;
 
     if (!hostent) return NULL;
 
-    sockaddr = talloc_zero(mem_ctx, struct sockaddr_storage);
+    switch(hostent->family) {
+        case AF_INET:
+            len = sizeof(struct sockaddr_in);
+            break;
+        case AF_INET6:
+            len = sizeof(struct sockaddr_in6);
+            break;
+        case AF_UNIX:
+            len = sizeof(struct sockaddr_un);
+            break;
+        default:
+            DEBUG(SSSDBG_CRIT_FAILURE,
+                  "Unknown address family %d\n", hostent->family);
+            return NULL;
+    }
+
+    sockaddr = (struct sockaddr *)talloc_zero(mem_ctx, struct sockaddr_storage);
     if (sockaddr == NULL) {
         DEBUG(SSSDBG_CRIT_FAILURE, "talloc_zero failed.\n");
         return NULL;
     }
 
+    if (sockaddr_len != NULL) {
+        *sockaddr_len = len;
+    }
```

Comment:
I would move this to the end of the function to avoid touching output arg in case function fails.

Reply:
Done in https://github.com/SSSD/sssd/pull/6484/commits/7e7cda40b02bd7102b74d39f904aa3b65bd5ae82